.github/workflows/node.js.yml

```diff
@@ -0,0 +1,61 @@
+name: CI
+
+on:
+    push:
+        branches:
+            - main
+    pull_request:
+
+jobs:
+    main:
+        runs-on: ubuntu-latest
+        if: ${{ github.event_name != 'pull_request' }}
```

Comment:
Creating a separate workflow for pull requests would be beneficial. You wouldn't have to add these ifs